int64[] id
int64[] temp
int64[] voltage
int64[] pos